### PR TITLE
Resurrected the ability to add and update roles

### PR DESCRIPTION
**Pull request by mkimberlin**

### CODE REVIEW

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 1]
```diff
@@ -21,32 +21,53 @@ import {
   CardContent,
   CardHeader,
```
Comment (pieperm) [inline web-ui/src/components/admin/roles/Roles.jsx:22]: Unused import

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 2]
```diff
@@ -21,32 +21,53 @@ import {
...
 } from "@mui/material";
 import PersonAddIcon from "@mui/icons-material/PersonAdd";
-import Autocomplete from '@mui/material/Autocomplete';
+import SearchIcon from "@mui/icons-material/Search";
+import AddIcon from "@mui/icons-material/Add";
 
 import { isArrayPresent } from './../../../helpers/checks';
 
 import "./Roles.css";
+import {selectProfile} from "../../../context/selectors";
```
Comment (pieperm) [inline web-ui/src/components/admin/roles/Roles.jsx:47]: Unused import

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 3]
```diff
@@ -149,152 +171,183 @@ const Roles = () => {
...
+            >
+              Add Role
+            </Button>
           </div>
-          {/* <Button color="primary" onClick={() => setShowAddRole(true)}>
-            Add New Role
-          </Button> */}
         </div>
-        <div className="roles-bot">
-          {uniqueRoles.map((role) =>
```
Comment (pieperm) [inline web-ui/src/components/admin/roles/Roles.jsx:182]: `uniqueRoles` is no longer being used, so the declaration on line 95 could be removed.

File: web-ui/src/components/admin/roles/Roles.jsx [hunk 4]
```diff
@@ -118,28 +139,29 @@ const Roles = () => {
...
-        payload: data,
+        type: SET_ROLES,
+        payload: updatedRoles,
       });
       window.snackDispatch({
         type: UPDATE_TOAST,
         payload: {
           severity: "success",
-          toast: `${member.name} added to ${selectedRole}s`,
+          toast: `Role Updated: ${role.role}`,
```
Comment (pieperm) [inline web-ui/src/components/admin/roles/Roles.jsx:164]: Do we want this toast to instead say "Role created"?

**Review by pieperm — COMMENTED** [review]
Discovered a bug: After creating a new role, members cannot be added it it.

Console error:
```
Uncaught (in promise) TypeError: role is undefined
    addToRole Roles.jsx:129
    onClick Roles.jsx:264
```

File: web-ui/src/components/admin/roles/Roles.jsx (hunk 4 above)
Comment (mkimberlin) [reply web-ui/src/components/admin/roles/Roles.jsx:160]: Nope. It's used for updates and additions.

**Comment by mkimberlin** [comment]
> Discovered a bug: After creating a new role, members cannot be added it it.
> 
> Console error:
> 
> ```
> Uncaught (in promise) TypeError: role is undefined
>     addToRole Roles.jsx:129
>     onClick Roles.jsx:264
> ```

Fixed.

**Review by pieperm — APPROVED** [review]

**Pull request merged**